src/ecs/input/system.rs

```diff
@@ -41,13 +41,13 @@ where
 
     fn run(&mut self, (input, mut handler, mut output): Self::SystemData) {
         match input.read(&mut self.reader) {
-            Ok(data) => for d in data {
+            Ok(EventReadData::Data(data)) => for d in data {
                 if let &Event::WindowEvent { ref event, .. } = d {
                     handler.send_event(event, &mut *output);
                 }
             },
 
```

Comment:
Might want a variant for EventReadData::Overflow that do the same things as above, and log a warning ?